.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,29 @@
+name: unit-test
+on:
+  pull_request:
+    branches:
+      - main
+permissions:
+  contents: read
+  pull-requests: write
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Go environment
+        uses: actions/setup-go@v5.3.0
+        with:
+          go-version-file: go.mod
+          cache: false
+      - name: Run unit tests
+        run: go test -race -parallel=$(nproc) -count=1 -timeout=30m -coverprofile cover.out -covermode=atomic -coverpkg=./... ./...
```

Comment:
wondering if we want to tighten this if we pay by the minute.  maybe 15m?  seems like it completes in 8m atm.